### ISSUE TITLE
all Nest devices non-response in Home app

### ISSUE THREAD

**Issue by kjustie** [chrisjshull/homebridge-nest#364]
A couple weeks ago all of my Nest devices -- 2 thermostats and a bunch of Protects -- became non-responsive in Home app on both iPhone and laptop; they all show in Home, but all display "no response."  All ARE responsive and DO work (in the case of the thermostats) in Homebridge app, itself.  I have 2 other Homebridge plugins, and those devices all work fine on all my devices.  Everything is up-to-date on devices and on Homebridge server (Mac), including all plugins, and restarting everything hasn't helped.

Today I deleted Homebridge from Home app on my phone and re-added it and all accessories.  It cycled through all the Nest devices, properly identifying them, allowing me to change names, locations, etc., but at the end of that process (and after restarting phone and waiting plenty long for things to update), all Nest devices are still "no response."  When I open a thermostat, all the "parts" are shown -- temp control, fan, eco mode -- but all are dimmed/non-responsive.  (I've not removed/re-added any of the Nest devices individually; since they're working fine in Homebridge, itself, and I've deleted and re-added the entire bridge and all accessories, it seems unlikely that would help.)

So, Homebridge, itself, appears to be working OK, and accessories associated with my other plugins are also working OK.  So any suggestions on what else to try with the Nest plugins to get them working in my Home app?

thanks

**Comment by adriancable** [comment]
@kjustie - please edit config.json and under "platform": "Nest" put:

`"options": [ "Debug.Verbose" ],`

Then please run: homebridge -D

Then wait for everything to load up, and email me the log at adrian.cable@gmail.com.

Since the Nest devices work OK in the Homebridge app, and your other devices work OK in HomeKit, my suspicion is one of your Nest devices has some property which is incompatible with HomeKit. (I try to check this, but probably haven't covered everything.) For example a very very very long name.

**Comment by kjustie** [comment]
Not sending the entire log (unless you still need it) because I saw that it included this:

[2/10/2021, 8:57:31 AM] [Nest] Auth failed: access token specified in Homebridge configuration rejected
[2/10/2021, 8:57:31 AM] [Nest] Unable to authenticate with Google/Nest.
[2/10/2021, 8:57:31 AM] [Nest] NOTE: Because we couldn't connect to the Nest service, your Nest devices in HomeKit will not be responsive.

That would seem to be the cause, but first, how can my access token be rejected when nothing's changed in my account?   And second, I was going to ask how my Nest devices could appear in Homebridge if it couldn't authenticate.....but I just found that while they appear, they actually aren't working, contrary to what I'd previously thought:  thermostat shows what appear to be accurate, live, changeable settings, but they're not "live" and temp changes are ignored.

So what's in Homebridge must just be old, cached, information?  

At this point, is it no longer possible to use Nest account instead of Google, as described on the homebridge-nest npm page? When I try the nest account token instructions, it eventually takes me to the "Migrate to Google Account" process, which I've been trying to avoid.

thanks

**Comment by adriancable** [comment]
@kjustie - I just tried the Nest account process as per the docs and it works fine. Can you describe in more detail where you get stuck?

I don't know why your token got rejected, but the most common reason is that you logged out of your Nest account in your browser (instead of just closing the tab). This invalidates the token.

**Comment by kjustie** [comment]
Sorry, my mistake: I was somehow distracted by the large Google login box -- and used that one instead -- and missed the small Nest login link.  Once I used the Nest link, it worked as expected, I got new token, put that in config file, and everything's working/updating correctly everywhere now.  

I couldn't tell you the last time I logged into my Nest account on my computer; certainly not in this timeframe, so that can't be how the previous token expired.  Is there anything I could have done in the Nest app (not in browser) on my phone that could have had the same effect?  It's much more likely I would have logged out/in there than in a browser on my computer.

Thanks for your help!
